### PR TITLE
Publish to ruby gems

### PR DESCRIPTION
**Pull request by ayinloya**
This introduces a github action to publish to ruby gems

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: Publish to RubyGems
+on:
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+* # should only run when the tags matches sematic versioning
```
Comment (vanshg) [inline .github/workflows/release.yml:5]: Is `+*` valid here/for GitHub Actions? It is invalid regex

Comment (vanshg) [reply .github/workflows/release.yml:5]: Also, do the `.`s need to be escaped? If this is parsed as regex, it would match any character

Comment (ayinloya) [reply .github/workflows/release.yml:5]: I noticed that the actions don't really use conventional regex
This make `.`s not been escaped and `+*` valid

[https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)

Comment (vanshg) [reply .github/workflows/release.yml:5]: Ah, I see. Although I'm not convinced on the validity of `+*`, I didn't see that in the docs, and I'm not sure what it means.

In the docs, they provide `v[12].[0-9]+.[0-9]+` as an example. What does the extra `*` at the end do?

Comment (ayinloya) [reply .github/workflows/release.yml:5]: the `*` supposed to match zero or more characters, but does not match the / character.
the `+` in `+*` has more of an effect on `[0-9]` than `*`
`[0-9]+*` makes it possible to match 15-beta, 23-patch, 12-pre-2, etc

**Review by vanshg — APPROVED** [review]

**Review by mldangelo — APPROVED** [review]
Thanks @ayinloya

**Pull request merged**